src/test/java/bio/terra/service/dataset/DatasetSchemaUpdateValidatorTest.java

```diff
@@ -1,45 +1,76 @@
 package bio.terra.service.dataset;
 
-import static bio.terra.common.fixtures.DatasetFixtures.*;
 import static org.hamcrest.CoreMatchers.containsString;
 import static org.hamcrest.MatcherAssert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
+import bio.terra.app.controller.ApiValidationExceptionHandler;
+import bio.terra.app.controller.DatasetsApiController;
 import bio.terra.common.TestUtils;
 import bio.terra.common.category.Unit;
 import bio.terra.common.fixtures.DatasetFixtures;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
 import bio.terra.model.ColumnModel;
 import bio.terra.model.DatasetSchemaUpdateModel;
 import bio.terra.model.DatasetSchemaUpdateModelChanges;
 import bio.terra.model.ErrorModel;
 import bio.terra.model.TableDataType;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.filedata.FileService;
+import bio.terra.service.job.JobService;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
 import java.util.List;
 import java.util.UUID;
 import org.apache.commons.lang3.StringUtils;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.http.MediaType;
 import org.springframework.mock.web.MockHttpServletResponse;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.web.servlet.MockMvc;
 import org.springframework.test.web.servlet.MvcResult;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
-@AutoConfigureMockMvc
 @ActiveProfiles({"google", "unittest"})
-@Category(Unit.class)
-public class DatasetSchemaUpdateRequestValidatorTest {
+@ContextConfiguration(
+    classes = {
+      DatasetSchemaUpdateValidator.class,
+      DatasetsApiController.class,
+      ApiValidationExceptionHandler.class
+    })
+@WebMvcTest
+@Tag(Unit.TAG)
+class DatasetSchemaUpdateValidatorTest {
 
   @Autowired private MockMvc mvc;
 
+  @MockBean private JobService jobService;
```

Comment:
This was required by the shift from using `SpringRunner` to `ContextConfiguration`. Before, the test was creating a spring application context such that all beans that weren't marked as `MockBean` in the test would be injected with real objects. This meant that the test was creating actual components for things like `IamService`, `DatasetService`, etc. This works but it means that a lot of code that's not being tested gets executed, and the test behavior could change due to code not being tested here.

Using `ContextConfiguration` flips things around so that _only_ the classes declared in its `classes` are real objects. Every other object must be mocked, or it it will trigger spring failure when it tries to create the context for the test.

Normally this is a good thing in that the test likely needs mocked objects anyway so it can verify or stub services that the test needs. The validator tests are a special case though. In this case, the tests require a fully mocked copy of `DatasetsApiController`, which ends up pulling in all the components that the controller needs.

Although now that I think about it, maybe this test could instead mock the controller itself, or use the controller's base class (`DatasetsApi`) instead. 🤔